contrib/bloop/src/mill.contrib.bloop/BloopModule.scala

```diff
@@ -0,0 +1,197 @@
+package mill.contrib.bloop
+
+import ammonite.ops._
+import bloop.config.Config.{File => BloopFile}
+import bloop.config.ConfigEncoderDecoders._
+import io.circe.Encoder
+import mill._
+import mill.define._
+import mill.eval.Evaluator
+import mill.modules.Jvm
+import mill.scalalib._
+import ujson.Visitor
+import upickle.default
+
+// TODO diamond problem is going to make this tricky. Using a self-type appears
```

Comment:
In what way is it problematic? Maybe we can show a code snippet upstream so that the macros are fixed to make it work?

Reply:
Good idea, I'll produce a minimal example to explain the issue ^^